pontoon/base/views.py

```diff
@@ -76,34 +81,81 @@ def home(request):
     return translate(request, locale, project.slug)
 
 
-def locale(request, locale, template='locale.html'):
-    """Locale view."""
-    log.debug("Locale view.")
+class TemplateModelView(TemplateView):
```

Comment:
I see now that you want send a message on 404. In that case, you'd want to sublcass `DetailView`, call it `PontoonDetailView` or something, and override `get_object` to catch 404s and send a message that subclasses can override.